speechbrain/utils/distributed.py

```diff
@@ -103,8 +94,10 @@ def main_process_only(function):
     @wraps(function)
     def main_proc_wrapped_func(*args, **kwargs):
         """This decorated function runs only if this is the main process."""
+        os.environ[MAIN_PROC_ENV] = "1"
         if if_main_process():
             return function(*args, **kwargs)
+        os.environ[MAIN_PROC_ENV] = "0"
```

Comment:
This line will be hit by a non main process and the variable will be changed, hence the ddp barrier afterward wouldn't work? no?

Reply:
Ok nevermind